engine/src/main/java/org/terasology/world/block/BlockArea.java

```diff
@@ -0,0 +1,622 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.world.block;
+
+import com.google.common.base.Preconditions;
+import org.joml.AABBf;
+import org.joml.AABBi;
+import org.joml.Math;
+import org.joml.Rectangled;
+import org.joml.Rectanglef;
+import org.joml.Rectanglei;
+import org.joml.RoundingMode;
+import org.joml.Vector2fc;
+import org.joml.Vector2i;
+import org.joml.Vector2ic;
+import org.joml.Vector3ic;
+
+import java.util.Objects;
+
+/**
+ * A bounded axis-aligned rectangle of blocks.
+ * <p>
+ * A block area is described and backed by an {@link org.joml.Rectanglei}
+ *
+ * @see BlockAreas
+ */
+public class BlockArea {
+    private final Vector2i min = new Vector2i(Integer.MAX_VALUE);
+    private final Vector2i max = new Vector2i(Integer.MIN_VALUE);
```

Comment:
@pollend Is this too much overhead keeping the vectors? Should I instead use just 4 plain integers?

Reply:
I would just use 4 integers.